### DISCUSSION TITLE
How to setup cache monitoring nicely

### DISCUSSION THREAD

**Discussion by yakobe** [getsentry/sentry-symfony#896] [Q&A]
Hello 👋 

I often look at the empty "caches" page in sentry and feel a bit disappointed that there are no metrics there. I have [read the docs](https://docs.sentry.io/platforms/php/guides/symfony/tracing/instrumentation/caches-module/) about manual instrumentation, and if feels very... well... manual 😅. It is a bit overkill to manually add it everywhere we get and put to the cache. Is there any best practices available so that we can configure it a bit more easily? Maybe using decorators or something? Perhaps some more docs would be possible?

Or even better... Is there any possibility that we could get a basic level integrated into sentry-symfony like the Laravel SDK has?

**Comment by cleptric** [comment]
I looked briefly at https://github.com/symfony/cache. While they offer some data collector, I don't think it's suitable for us, as the telemetry created is not like we would need it.
So it would be likely back to decorating the adapter. We do not have immediate plans do this, but I would be happy to review a PR.

**Reply by yakobe** [reply]
Thanks for getting back to me @cleptric. Any idea what is missing from the symfont cache implementation to get it working? Maybe i can try to look into it. I tried looking into how the laravel one is implemented ([i think this is it](https://github.com/getsentry/sentry-laravel/blob/master/src/Sentry/Laravel/Features/CacheIntegration.php)) and it seems to be based around some events that I'm guessing symfony does not provide?

In the mean time, I will probably try to write a decorator for now to see if i can get some data 😄

**Reply by cleptric** [reply]
Oh, I stand corrected 😬  We actually already offer support for some basic cache tracing, https://github.com/getsentry/sentry-symfony/tree/master/src/Tracing/Cache.
Ok, let me take a look and I get back to you. I think with some minor adjustments we can already get some data to show up on the Cache module page.
